arxiv/arxiv.py

```diff
@@ -208,7 +209,8 @@ def download_pdf(self, dirpath: str = "./", filename: str = "") -> str:
         if not filename:
             filename = self._get_default_filename()
         path = os.path.join(dirpath, filename)
-        written_path, _ = urlretrieve(self.pdf_url, path)
+        # FIXME: `self.pdf_url` is possibly `None` and following code raises `TypeError`.
+        written_path, _ = urlretrieve(self.pdf_url, path)  # type: ignore[arg-type]
```

Comment:
`arxiv.Result._get_pdf_url([])` returns `None` and it raises `TypeError` in this line.